### ISSUE TITLE
How to import .sql backup into new postgres

### ISSUE THREAD

**Issue by wneiner** [boschkundendienst/guacamole-docker-compose#30]
Hello,  I had a previous oznu guacamole setup.  It is no longer maintained.  I have the backup.sql from my postgres there.  Where would I find the postgres.sql in your guacamole-docker-compose?

If I can understand the location, I can use this to import it. 

      psql -U username -d dbname < filename.sql

Any help is really appreciated!

**Comment by boschkundendienst** [comment]
Hi,

please understand that this is more of a postgres question, rather unrelated to my project here. Nevertheless some info.

In my compose file I use the official postgres image from [here](https://hub.docker.com/_/postgres). There is also a very good documentation available on this page.

Additionally (if the container is running), you can e.g. connect into the container with the command `docker exec -it <container-id> /bin/bash` to directly execute commands towards postgres. (replace `<container-id>` with the id of your postgres container)


The `./data` folder from "outside" should be located within the container at `/var/lib/postgresql/data`.

So you can e.g. put your backup in `./data` and then address it in the container under `/var/lib/postgresql/data`.

`psql` is included in the postgres container.

Unfortunately I cannot say much more about this. It may also be necessary (if something has changed in the database schema since your last backup in Guacamole) to run SQL update scripts from Guacamole after you have imported your database.

Since the update files from Guacamole to update the DB scheme are located within the guacamole/guacamole container it might be necessary to copy them into the postgres container or access the postgres container from within a shell out of guacamole/guacamole. So it can get quite complicated.

I would say better help is available on the official [mailing list of the Guacamole project](https://guacamole.apache.org/support/). Your problem actually has nothing to do with my project and the people on the [mailing list](https://guacamole.apache.org/support/) are really helpful and can you walk exactly to a possible solution.

And what you also can do is to install guacamole natively without docker and do the upgrade/restore there.

I hope I could help a bit.

Regards

Peter

**Comment by comsupany** [comment]
Hi!  I am new to all of this but I have a question.  Is there a way to change ports 8443 and 443 to 8080?  I'm having a problem with my browsers being confused.  I can eventually connect to http://myip:8443 but it doesn't always work.  There is probably something wrong with the certificate but I am only testing internally and don't need a certificate.  I changed the yaml file but that didn't work for me.  Thank you!

**Comment by boschkundendienst** [comment]
Please read the docs and anyways, please open a new issue if it is not related to the previous question.

**Comment by comsupany** [comment]
> Please read the docs and anyways, please open a new issue if it is not related to the previous question.

Yeah my mistake posting here but I found nothing in the docs to help me.  Can you help a newbie?

**Comment by boschkundendienst** [comment]
Hi again.

Some clarification here. When the containers are running doing a `docker ps -a` shows that for Guacamole (if you used my scripts) in the section `PORTS` you have something like this:

```
guacamole/guacamole	49153 (outside)	8080 (inside)
guacamole/guacd		4822  (inside only)
postgres:13.4		5432/tcp (inside only)
nginx			8443  (outside) 80/443 inside
```
So first of all nothing (on the outside) is listening on port `443` on the computer running my Guacamole setup. Port 443 is only used inside the nginx container but only exposed on port `8443` to the outside world. So when you write "Is there a way to change ports 8443 and 443 to 8080?" the only question that makes sence is to ask "Is there a way to change port 8443 to 8080?".

And this is possible. The relevant section in the `docker-compose.yml` file (nearly the end) looks like this:

```
########### optional ##############
  # nginx
  nginx:
   container_name: nginx_guacamole_compose
   restart: always
   image: nginx
   volumes:
   - ./nginx/ssl/self.cert:/etc/nginx/ssl/self.cert:ro
   - ./nginx/ssl/self-ssl.key:/etc/nginx/ssl/self-ssl.key:ro
   - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
   - ./nginx/mysite.template:/etc/nginx/conf.d/default.conf:ro
   ports:
   - 8443:443
   links:
   - guacamole
   networks:
     guacnetwork_compose:
   # run nginx
   command: /bin/bash -c "nginx -g 'daemon off;'"
# nginx-debug-mode
#   command: /bin/bash -c "nginx-debug -g 'daemon off;'"
```

Just change the `8443` on the left side of the line `   - 8443:443` to whatever port you like (must be free to use on your machine) and then you can connect to
`https://<your-ip>:<port you configured>/` to open Guacamole.

Everything else should only be changed if you know what your are doing and I can not support that.

Of course, feel free to read the docs at the Guacamole Homepage ([Installing Guacamole with Docker](https://guacamole.apache.org/doc/gug/guacamole-docker.html)).
It also explains how all the components work together and how you can setup Guacamole with Docker.

This was actually what I have read when creating this project here. I just wanted to have a quick&dirty way to spin up Guacamole for testing.

I hope I could help a bit.

Regards

Peter

**Comment by wneiner** [comment]
@boschkundendienst After a while of not being able to come back to this project, I just wanted to reply and say thank you.  I got it figured out and you were key in guiding me there.  I actually learned a lot with your help, now I am down a whole new rabbit hole.

For others who could use the help.  As stated this question wasn't directly related to this project but instead Postgres and how to interact with it in the container.

To backup my current Postgres I used this command:

    `sudo docker exec CONTAINER_ID pg_dump -U POSTGRES_USER POSTGRES_DB > backup.sql`

I then needed to put that inside my container:

    `sudo docker cp /home/HOMEDIRECTORY/backup.sql CONTAINER_ID:/`

Then I needed to exec into my container:

    `sudo docker exec -it CONTAINER_ID /bin/bash`

Last step is to import the backup into the new Postgres Container:

    `psql -U POSTGRES_USER -d POSTGRES_DB < backup.sql`

Thank you again!
